test/distributed/_composable/fsdp/test_fully_shard_training.py

```diff
@@ -275,7 +274,8 @@ def _test_train_parity_single_group(self, lin_shapes: List[Tuple[int, int]]):
             self.assertEqual(losses[0], losses[1])
 
     @skip_if_lt_x_gpu(2)
```

Comment:
we can support `reshard_after_forward=2` with `torch._dynamo.config.skip_fsdp_hooks=True`

it's a good time to merge `test_train_parity_multi_group_eager`, `_compile`, `cpu_offload_eager` into `test_train_parity_multi_group`.

Reply:
I recall that if we merged `cpu_offload_eager` with the normal eager test, sometimes the test becomes flaky for unknown reasons.

Reply:
It might be good to keep the CPU offloading one separate.

Reply:
will keep cpu offloading as a separate test